interfaces/builtin/desktop.go

```diff
@@ -216,8 +219,33 @@ func (iface *desktopInterface) AutoConnect(*interfaces.Plug, *interfaces.Slot) b
 	return true
 }
 
+func (iface *desktopInterface) fontconfigDirs() []string {
+	return []string{
+		dirs.SystemFontsDir,
+		dirs.SystemLocalFontsDir,
+		dirs.SystemFontconfigCacheDir,
+	}
+}
+
 func (iface *desktopInterface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
 	spec.AddSnippet(desktopConnectedPlugAppArmor)
+
+	if !release.OnClassic {
+		// We only need the font mount rules on classic systems
+		return nil
+	}
+
+	for _, dir := range iface.fontconfigDirs() {
```

Comment:
As much as I do not like templates, this one looks like a nice candidate:
```
var desktopNSTemplate = template.Must(template.New("desktopNS").Parse(`  # Read-only access to {{ .Target }}
  mount options=(bind) {{ .Source }}/ -> {{ .Target }}/,
  remount options=(bind, ro) {{ .Target }}/,
  umount {{ .Target }}/,

`))
...
func (iface *desktopInterface) AppArmorConnectedPlug(....) {
	var buf bytes.Buffer
	for _, dir := range iface.fontconfigDirs() {
		desktopNSTemplate.Execute(&buf, struct{
			Target string
			Source string
		}{"/var/lib/snapd/hostfs" + dir, dirs.StripRootDir(dir)})
	}
	spec.AddUpdateNS(buf.String())
	return nil
}
```
See https://play.golang.org/p/zE-iz0p9YIz

Reply:
This can be done in a follow-up, I won't block on that.